Sanic Webserver Support (#724)
strawberry-graphql/strawberry: docs/integrations/sanic.md

---
title: Sanic
---

# Sanic

Strawberry comes with a basic Sanic integration. It provides a view that you can

> use to serve your GraphQL schema:

Author: Done.

```python
from strawberry.sanic.views import GraphQLView

from api.schema import Schema

app = Sanic(__name__)

app.add_route(
    GraphQLView.as_view(schema=schema, graphiql=True),
    "/graphql",
)
```

## Options

The `GraphQLView` accepts two options at the moment:

> - schema: mandatory, the schema created by `strawberry.Schema`.
> - graphiql: optional, defaults to `True`, whether to enable the GraphiQL
>   interface.

Reviewer: ```suggestion
- `schema`: mandatory, the schema created by `strawberry.Schema`.
- `graphiql`: optional, defaults to `True`, whether to enable the GraphiQL
  interface.
```

## Extending the view

> We allow to extend the base `GraphQLView`, by overriding the following methods:

Reviewer: ```suggestion
The base `GraphQLView` class can be extended by overriding the following methods:
```

- `get_context(self) -> Any`
- `get_root_value(self) -> Any`
- `process_result(self, result: ExecutionResult) -> GraphQLHTTPResponse`

## get_context

> `get_context` allows to provide a custom context object that can be used in your
> resolver. You can return anything here, by default we return a dictionary with
> the request.

Reviewer: ```suggestion
By overriding `GraphQLView.get_context` you can provide a custom context object for
your resolvers. You can return anything here; by default GraphQLView returns a
dictionary with the request.
```

```python
class MyGraphQLView(GraphQLView):
    def get_context(self, request) -> Any:
        return {"example": 1}

@strawberry.type
class Query:
    @strawberry.field
    def example(self, info: Info) -> str:
        return str(info.context["example"])
```

> Here we are returning a custom context dictionary that contains only one item
> called "example".

Reviewer: ```suggestion
Here we are returning a custom context dictionary that contains only one item
called `"example"`.
```

> Then we use the context in a resolver, the resolver will return "1" in this
> case.

Reviewer: ```suggestion
Then we can use the context in a resolver. In this case the resolver will return `1`.
```

Reviewer: I would also rearrange these code snippets to be a bit more interspersed with the descriptions

## get_root_value

> `get_root_value` allows to provide a custom root value for your schema, this is
> probably not used a lot but it might be useful in certain situations.

Reviewer: ```suggestion
By overriding `GraphQLView.get_root_value` you can provide a custom root value for your
schema. This is probably not used a lot but it might be useful in certain situations.
```

Here's an example:

```python
class MyGraphQLView(GraphQLView):
    def get_root_value(self) -> Any:
        return Query(name="Patrick")

@strawberry.type
class Query:
    name: str
```

> Here we are returning a Query where the name is "Patrick", so we when requesting
> the field name we'll return "Patrick" in this case.

Reviewer: ```suggestion
Here we configure a Query where requesting the `name` field will return `"Patrick"`
through the custom root value.
```

## process_result

> `process_result` allows to customize and/or process results before they are sent
> to the clients. This can be useful logging errors or hiding them (for example to
> hide internal exceptions).

Reviewer: ```suggestion
By overriding `GraphQLView.process_result` you can customize and/or process results
before they are sent to a client. This can be useful for logging errors, or even hiding
them (for example to hide internal exceptions).
```
